src/App.jsx

```diff
@@ -14,45 +16,95 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_BY_NAME = 'name';
+const SORT_BY_LENGTH = 'length';
+
+function sortGoods(goods, sortType, isReversed) {
+  const preparedGoods = [...goods];
+
+  if (sortType) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortType) {
+        case SORT_BY_LENGTH:
+          return good1.length - good2.length;
+
+        case SORT_BY_NAME:
+          return good1.localeCompare(good2);
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+export const App = () => {
+  const [sortType, setSortType] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
+  const visibleGoods = sortGoods(goodsFromServer, sortType, isReversed);
+  const reset = () => {
+    setSortType('');
+    setIsReversed(false);
+  };
+
+  const isShowReset = sortType || isReversed;
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={cn('button', 'is-info',
+            { 'is-light': sortType !== SORT_BY_NAME })}
+          onClick={() => setSortType(SORT_BY_NAME)}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          className={cn('button', 'is-success',
+            { 'is-light': sortType !== SORT_BY_LENGTH })}
+          onClick={() => setSortType(SORT_BY_LENGTH)}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={cn('button', 'is-warning',
+            { 'is-light': !isReversed })}
+          onClick={() => setIsReversed(!isReversed)}
+        >
+          Reverse
+        </button>
+
+        {isShowReset && (
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={() => {
+              reset();
+            }}
```

Comment:
```suggestion
            onClick={reset}
```

can be more simplify